### PR TITLE
Address the remaining 'low-hanging fruit' issues

### PR DESCRIPTION
**Pull request by nrc**

### CODE REVIEW

File: guide/items.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 ## Items
 
+`extern crate` statements must be first in a file. They must be ordered
```
Comment (pyfisch) [inline guide/items.md]: There is already an [Extern crate] section. These two sentences should probably be placed there.

[Extern crate]: https://github.com/rust-lang-nursery/fmt-rfcs/blob/86720261f2bd1477062bf584ddcae6c859f90bfd/guide/items.md#extern-crate

File: guide/items.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+alphabetically.
+
+`use` statements, and module *declarations* (`mod foo;`, not `mod { ... }`)
+must come before other items. We recommend that imports come before module
+declarations; if imports and modules are separated, then they should be ordered
```
Comment (pyfisch) [inline guide/items.md]: What happens to relative imports like `super::` and `self::` should they be sorted under S? Also many people prefer the three sections rule. Maybe allow this by stating that there may be different sections of imports each separated by a blank line with the lines in a section sorted alphabetically.

**Comment by topecongiro** [comment]
When reordering `mod;` or `extern crate` alphabetically, we must respect attributes on them (cc https://github.com/rust-lang-nursery/rustfmt/issues/2399, https://github.com/rust-lang/rust/issues/32911). AFAIK the behavior of these items is order dependent only when they have `#[macro_use]`, but I am not sure.

**Comment by matklad** [comment]
Curious: is the motivation behind ordering `mod` declarations strong, or it's just "doesn't matter much, but some ordering is better than none"?

I personally feel pretty strongly that, by default, the order of mod declarations should be preserved. 

When a new person reads the code-base for the first time, ordering of mod declarations can provide a very useful hints as to in which order the code should be read. 

To me, sorting mod declarations seems close to sorting struct definitions than to sorting imports.

**Pull request merged**